lib/formdata/0.0.2/index.js

```diff
@@ -0,0 +1,93 @@
+/*
+ * FormData polyfill for k6
+ * Copyright (C) 2021 Load Impact
+ * License: MIT
+ *
+ * This simplifies the creation of multipart/form-data requests from k6 scripts.
+ * It was adapted from the original version by Rob Wu[1] to remove references of
+ * XMLHttpRequest and File related code which isn't supported in k6.
+ *
+ * [1]: https://gist.github.com/Rob--W/8b5adedd84c0d36aba64
+ **/
+
+if (exports.FormData) {
+  // Don't replace FormData if it already exists
+  return;
+}
+
+// Export variable to the global scope
+exports.FormData = FormData;
+
+function FormData() {
+  // Force a Constructor
+  if (!(this instanceof FormData)) return new FormData();
+  // Generate a random boundary - This must be unique with respect to the
+  // form's contents.
+  this.boundary = '------RWWorkerFormDataBoundary' + Math.random().toString(36);
+  this.parts = [];
+
+  /**
+   * Internal method. Convert input to a byte array.
+   * @param inp String | ArrayBuffer | Uint8Array  Input
+   */
+  this.__toByteArray = function(inp) {
+    var arr = [];
+    var i = 0, len;
+    if (typeof inp == 'string') {
+      for (len = inp.length; i < len; ++i)
+        arr.push(inp.charCodeAt(i) & 0xff);
+    } else if (inp && inp.byteLength) {/*If ArrayBuffer or typed array */
+      if (!('byteOffset' in inp))   /* If ArrayBuffer, wrap in view */
+        inp = new Uint8Array(inp);
+      for (len = inp.byteLength; i < len; ++i)
+        arr.push(inp[i] & 0xff);
+    }
+    return arr;
+  };
+}
+
+/**
+ * @param fieldName          String                    Form field name
+ * @param data               object|string             An object or string field value.
+ *
+ * If data is an object, it should match the structure of k6's http.FileData
+ * object (returned by http.file()) and consist of:
+ * @param data.data          String|Array|ArrayBuffer  File data
+ * @param data.filename      String                    Optional file name
+ * @param data.content_type  String                    Optional content type, default is application/octet-stream
+ **/
+FormData.prototype.append = function(fieldName, data) {
+  if (arguments.length < 2) {
+    throw new SyntaxError('Not enough arguments');
+  }
+  var file = data;
+  if (typeof data == 'string') {
+    file = {data: data};
```

Comment:
It used to be possible to specify content_type and filename when passing data as a string. It should probably still be possible? It is being used in the testbuilder today.

Maybe we can infer text/plain as a default when data is passed as string too?

![Screenshot 2021-03-15 at 15 40 50](https://user-images.githubusercontent.com/1597337/111171063-dd5e2f00-85a4-11eb-980c-7c50a893c818.png)

Reply:
Yeah, that's still possible.

As for inferring `text/plain`... maybe? I'm a bit apprehensive about implicit behavior like this, though in this case it seems harmless and probably what the user wants. OK, I'll add it.

Reply:
Yeah but the interface changed it looks like, or maybe I'm confused? But it looks like:

0.0.1
```
let filedata = { data: "foo", content_type: "text/plain" }
this.parts.push({field: fieldName, file: filedata})
// {field: fieldName, file: { data: "foo", content_type: "text/plain" }}
```

0.0.2
```
let data = { data: "123", content_type: "text/plain" }

  if (typeof data == 'string') {
    file = {data: data};
    // { data: { data: "123", content_type: "text/plain" } }
  }
  
  this.parts.push({field: fieldName, file: file});
```

Reply:
I ah see 🤦  it IS just me being confused 👍